base/logger_file.go

```diff
@@ -140,6 +143,20 @@ func (lfc *FileLoggerConfig) init(level LogLevel, name string, logFilePath strin
 		lfc.CollationBufferSize = &bufferSize
 	}
 
+	if lfc.Rotation.RotatedLogsSizeLimit == 0 {
+		lfc.Rotation.RotatedLogsSizeLimit = defaultCumulativeMaxSizeBeforeDeletion
+	}
+
+	ticker := time.NewTicker(time.Hour)
+	go func() {
+		for {
+			select {
+			case <-ticker.C:
+				runLogDeletion(logFilePath, level.String(), lfc.Rotation.RotatedLogsSizeLimit)
```

Comment:
We should add the threshold we talked about to this size limit, so that we're not constantly deleting stuff as soon as a new log is rotated.